src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +269,23 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park (and unpark? repark?) to error. The error
+        contains a reference to the task sent as a parameter."""
+        self.broken_by = task
+
+        # TODO: is there any reason to use self._pop_several?
```

Comment:
Probably some thin veneer of thread safety? Or guaranteeing that we'll remove every task we raise an error in, even if somehow `_core.reschedule` raises an error.

Neither is really something we care about but probably worth using cause it's a simple drop in replacement anyways.

Reply:
`unpark()` above would have the same theoretical issue, if `reschedule()` failed it could drop tasks too. Probably not something to worry about, if that fails we've got bigger problems.